src/subcommands/deployments.ts

```diff
@@ -334,6 +376,183 @@ async function deleteDeployment(args: Args): Promise<void> {
+  const prod = args.prod ?? alreadyProd;
+  const prodDatabase = databases?.find((database) =>
+    deploymentDatabaseEnv(project, database) === "Production"
+  );
+  const previewDatabase = databases?.find((database) =>
+    deploymentDatabaseEnv(project, database) === "Preview"
+  );
+  let db;
+  switch (args.db?.toLowerCase().trim()) {
+    case "prod":
+    case "production": {
+      if (!prodDatabase) {
+        spinnerPrep.fail(
+          `Project '${project.name}' does not have a production database`,
+        );
+        return Deno.exit(1);
+      }
+      db = prodDatabase.databaseId;
+      break;
+    }
+    case "preview": {
+      if (!previewDatabase) {
+        spinnerPrep.fail(
+          `Project '${project.name}' does not have a preview database`,
+        );
+        return Deno.exit(1);
+      }
+      db = previewDatabase.databaseId;
+      break;
+    }
+    default:
+      db = args.db;
+  }
+
+  if (!db) {
+    // For GitHub deployments, Deploy assigns the branch database also during redeployment
+    // Unless the user is explicit about the db, we want to maintain the invariant status == databaseEnv
+    if (prod) {
+      db = prodDatabase?.databaseId;
+    } else {
+      db = previewDatabase?.databaseId;
+    }
+  }
+
+  const envVarsToAdd = await envVarsFromArgs(args) || {};
+  const addedEnvs = Object.keys(envVarsToAdd);
+  // If the redeployment sets some env vars, the remaining env vars in the deployment are deleted
+  const envVarsToRemove = build.deployment && addedEnvs.length > 0
+    ? Object.fromEntries(
+      build.deployment.envVars
+        .filter((env) => !addedEnvs.includes(env))
+        // HOME is always set by Deno Deploy
+        .filter((env) => env !== "HOME")
+        .map((key) => [key, null]),
+    )
+    : {};
+  const removedEnvs = Object.keys(envVarsToRemove);
+
+  const envVars = {
+    ...envVarsToAdd,
+    ...envVarsToRemove,
+  };
+
+  spinnerPrep.succeed(
+    `Redeployment of deployment '${deploymentId}' is ready to begin:`,
+  );
+
+  const domainMappingDescription = prod
+    ? "The new deployment will be the new production deployment"
+    : "The new deployment will be a preview deployment";
+
+  wait({ text: "", indent: 3 }).start().info(domainMappingDescription);
+  if (db) {
+    const dbTag = db === prodDatabase?.databaseId
+      ? "production"
+      : db === previewDatabase?.databaseId
+      ? "preview"
+      : "custom";
+    wait({ text: "", indent: 3 }).start().info(
+      `The new deployment will use the ${dbTag} database '${db}'`,
+    );
+  }
+
+  if (addedEnvs.length === 1) {
+    wait({ text: "", indent: 3 }).start().info(
+      `The new deployment will use the env variable ${addedEnvs[0]}`,
+    );
+  } else if (addedEnvs.length > 1) {
+    wait({ text: "", indent: 3 }).start().info(
+      `The new deployment will use the env variables ${
+        addedEnvs.slice(0, -1).join(", ")
+      } and ${addedEnvs.at(-1)}`,
```

Comment:
It would be good to have this formatting logic as a separate utility function

Reply:
In this case I think the drawbacks of extracting this in a separate function outweight the benefits. The boilerplate that we would be saving is minimal, and it would fragment the linearity of the render code.

Reply:
OK that's fine too, I would personally prefer extracting things into separate functions especially when they are referential transparent (thus easy to unit test), but it's just a matter of preference and not a big deal 👍